climpyrical/tests/test_spytial.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+import numpy as np
+import climpyrical.spytialProcess as sp
+from climpyrical.gridding import flatten_coords
+
+N = 10
+xx, yy = np.meshgrid(np.linspace(0, 50, N), np.linspace(-25, 25, N))
+nxx, nyy = flatten_coords(np.linspace(0, 50, N), np.linspace(-25, 25, N))
+coords = np.stack([nxx, nyy])
+z = np.sin(xx ** 2 + yy ** 2) / (xx ** 2 + yy ** 2)
+z = z.flatten()
+
+new_N = 3 * N
+
+
+@pytest.mark.parametrize(
+    "latlon, z, nx, ny, xy, distance, variogram_model, error",
+    [
+        (coords, z, 3 * N, 3 * N, (1, 2), "geo", "exponential", None),
+        ("blargh", z, 3 * N, 3 * N, (1, 2), "geo", "exponential", TypeError),
```

Comment:
Ahhhh, again, a large percentage of your test cases (90%?) is for `TypeError`s/`ValueError`s. If you can automate the type checking, then that eliminates a whole class of automated tests that you have to write yourself :)